### ISSUE TITLE
Treasury Recuring Deposits

### ISSUE THREAD

**Issue by Hornan7** [IntersectMBO/cardano-ledger#4000]
## Context:
- We made a governance to withdraw 3 millions from the treasury 
- The action was ratified on Epoch 212 and enacted on Epoch 213 which is when I received the 3 millions, (With the help of our DRep that has over 51% of total voting stake + ourselves as the Constitutional Committee)

TxId: `d9fe066acdb60da93f0268baf337807122428eece43d898f8452e9d566d96ba9`
Index: `0`

## Issue:
As much as I like the treasury's devotion to my cause, it seems that its sending me recurring deposits of 3 millions every epochs
.....which is now at a total of 24 Millions 8 Epochs later. (Epoch 221)
```
[
    {
        "address": "stake_test1uqvj4r4h0fh5qhvpjmwedkcnj5xysvq2plutdapw0yq69mccrcnd7",
        "delegationDeposit": 2000000,
        "rewardAccountBalance": 24000368729198,
        "stakeDelegation": "pool1knw0wt62r2yv7v455j2p94lx98g9ltjv3a3aq8m4urlncwftf6e",
        "voteDelegation": "keyHash-c1ba49d52822bc4ef30cbf77060251668f1a6ef15ca46d18f76cc758"
    }
]
```
To be sure my pool wasn't involved, and the rewards i get from it, I did the same thing from an un-delegated registered stake address.

TxId: `7277b7cd4df7674ecc437c0c9fabb33bdbf9c60a13898a3b31729f27c450032a`
Index: `0`

This time with 64 millions and the second epoch after enactment it went up to 128 millions.
I spreaded it across 5 unregistered wallet for now to make sure to preserve the chain from its vote/stake influence. 
```
                           TxHash                                 TxIx        Amount
--------------------------------------------------------------------------------------
5d2e8ab96595145f09efc9d03a787fd1aeaeca94decaad6fa0167031c7753266     4        72000997370689 lovelace + TxOutDatumNone
                           TxHash                                 TxIx        Amount
--------------------------------------------------------------------------------------
5d2e8ab96595145f09efc9d03a787fd1aeaeca94decaad6fa0167031c7753266     2        16000000000000 lovelace + TxOutDatumNone
                           TxHash                                 TxIx        Amount
--------------------------------------------------------------------------------------
5d2e8ab96595145f09efc9d03a787fd1aeaeca94decaad6fa0167031c7753266     3        16000000000000 lovelace + TxOutDatumNone
                           TxHash                                 TxIx        Amount
--------------------------------------------------------------------------------------
5d2e8ab96595145f09efc9d03a787fd1aeaeca94decaad6fa0167031c7753266     1        16000000000000 lovelace + TxOutDatumNone
                           TxHash                                 TxIx        Amount
--------------------------------------------------------------------------------------
5d2e8ab96595145f09efc9d03a787fd1aeaeca94decaad6fa0167031c7753266     0        8000000000000 lovelace + TxOutDatumNone
```

**Comment by disassembler** [comment]
Known issue and already fixed in ledger PR in final review:
https://github.com/IntersectMBO/cardano-ledger/pull/3978

If it doesn't cause too much work we'll try to get this in 8.8!

**Comment by lehins** [comment]
This problem has been reported a couple of times in the last month. There was a lot of changes and fixes that were happening in the area of enactment since then. I was pretty confident that the issue has been fixed for a while, but just to be safe I did want to confirm that it was fixed. Therefore I was able to reproduce the failure with the test implemented in #4004 and verified that the issue was fixed back in #3855

The #3978 PR that has just been merged improves further the functionality in this area, by ensuring that the hierarchy of proposals are properly cleaned up as well.
